c++/src/kj/async-io-unix.c++

```diff
@@ -1343,6 +1343,15 @@ public:
 
     if (newFd >= 0) {
       kj::OwnFd ownFd(newFd);
+      
+      // On macOS, accept() can return a valid file descriptor with addrlen == 0 when a 
```

Comment:
I can wrap this in an ifdef APPLE if you prefer but this seemed like a generally good idea even for linux so I left this unguarded.